app/scripts/services/quota.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .factory("QuotaService", function(APIService,
+                                    $filter,
+                                    $q,
+                                    DataService) {
+
+    var isBestEffort = $filter('isBestEffortPod');
+    var isTerminating = $filter('isTerminatingPod');
+    var isNil = $filter('isNil');
+
+    var filterQuotasForPod = function(pod, quotas) {
+      var bestEffortPod = isBestEffort(pod);
+      var terminatingPod = isTerminating(pod);
+      return _.filter(quotas, function(quota){
+        // A quota matches a pod if all scopes match the pod
+        // Break out early if we find any scope that does not match
+        return !_.some(quota.spec.quota ? quota.spec.quota.scopes : quota.spec.scopes, function(scope) {
+          var matches = true;
+          switch(scope) {
+            case "Terminating":
+              matches = terminatingPod;
+              break;
+            case "NotTerminating":
+              matches = !terminatingPod;
+              break;
+            case "BestEffort":
+              matches = bestEffortPod;
+              break;
+            case "NotBestEffort":
+              matches = !bestEffortPod;
+              break;
+          }
+          return !matches;
+        });
+      });
+    };
+
+    var filterQuotasForResource = function(resource, quotas) {
+      if (resource.kind === 'Pod') {
+        return filterQuotasForPod(resource, quotas);
+      }
+      // We plan on having other resources that scopes will affect
+      return quotas;
+    };
+
+    var humanizeQuotaResource = $filter('humanizeQuotaResource');
+    var humanizeKind = $filter('humanizeKind');
+
+    var getResourceLimitWarnings = function(resource, quota){
```

Comment:
`getResourceLimitAlerts`? Since some of these are errors and we return an alerts array.